trie/zk_trie_impl.go

```diff
@@ -72,14 +82,23 @@ func NewZkTrieImplWithRoot(storage ZktrieDatabase, root *zkt.Hash, maxLevels int
 }
 
 // Root returns the MerkleRoot
-func (mt *ZkTrieImpl) Root() *zkt.Hash {
+func (mt *ZkTrieImpl) Root() (*zkt.Hash, error) {
+	hashedDirtyStorage := make(map[zkt.Hash]*Node)
+	rootKey, err := mt.commit(mt.rootKey, hashedDirtyStorage, new(sync.Mutex))
+	if err != nil {
+		return nil, err
+	}
+
+	mt.rootKey = rootKey
```

Comment:
> "drop" the dirty data and reverse trie to the original state easily

I don't think there is a function that allow the caller to do that, no? If a user needs to do that, all they need to do is create the trie with `NewZkTrieImplWithRoot` like they did initially. Since all changes are in-memory until commit is called, there is no need to do anything complicated to `rewind` the trie

Reply:
ok